src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateHistory = [];
+  let currentState = {};
+
+  Object.assign(currentState, state);
```

Comment:
You're already using Object.assign, but it's better to use destructuring to copy the initial state.

```suggestion
  let currentState = { ...state };
```